src/views/migrationpolicies/list/components/MigrationPoliciesEmptyState/MigrationPoliciesEmptyState.tsx

```diff
@@ -39,7 +46,11 @@ const MigrationPoliciesEmptyState: FC = () => {
       </EmptyStateBody>
       <EmptyStateFooter>
         <EmptyStateActions>
-          <MigrationPoliciesCreateButton />
+          {isUnprivilegedUser ? (
+            <MigrationPoliciesCreateButtonDisabled />
+          ) : (
+            <MigrationPoliciesCreateButton />
+          )}
```

Comment:
why not using 1 component for the button and let this component check for privileges?

Reply:
Because code is more readable now and this can be changed very soon, too, as the discussions about future of MigrationPolicies is still happening and different approaches occur, so it would be much easier to delete the unnecessary part of the code, once another change will be proposed (it is probable).

Reply:
It seems to me like a ternary inside a return of a component is much less readily and increases file size; this can be combined into a single component of MigrationPoliciesCreateButton

Reply:
So I will have to put ternary operator inside another component :D I don't see any difference. Also I will have to propagate another prop there, unnecessary.

Reply:
well no, if you build it correctly you won't need the ternary

Reply:
and the component check the the privileges so no need for another prop

Reply:
Well in this file the component does not check exactly what we need. We need to check the error message, unfortunately. Also I am building the logic on top of what was already built there (not ideal) and according to actual conditions (changing decisions of various people here and there and too often). Maybe I should have added a comment to the code, to make it more clear and provide more info about why this way.

Reply:
> Well I am still not convinced. In this file the component does not check exactly what we need. We need to check the error message specifically, unfortunately. Also I am building the logic on top of what was already built there (not ideal) and according to actual conditions (changing decisions of various people here and there and too often). Maybe I should have added a comment to the code, to make it more clear and provide more info about why this way.

1. Yes, u need to add it.
2. We don't need to check error messages, and we don't do it anywhere else in our code base, as messages can be changed. They aren't part of the API contract; they are simply strings.
3. Don't just add logic and make it less maintainable, readable, and scalable. Every touch in code should improve, not just add things because it's the assignment.
4. What do u mean by changing the decisions of various people? what was implemented before won't mean it's what we want now. the product is getting bigger, more requirements are added all the time, and we need to scale and build accordingly; things are always changing and will continue to change; what was right a month ago may not be right two months ahead, according to the road map. it's ok to change the code that others wrote in order to get the task done, but we need to make the code better, not harder to maintain and read.

If you still don't want to make the right changes and would like to continue to discuss this, you can reach me privately.

Reply:
Of course I do want to make the right changes. But it seems we have very different opinion on what is right actually, and it seems you don't understand many reasons why and what I had to face to make the decisions I made. Will talk to you later.